www/js/services/transaction_service.js

```diff
@@ -72,5 +76,34 @@ app.service('TransactionService', function($q, UserService, RequestParameterBuil
     return this.charge(amount * -1, description);
   };
 
+  TransactionService.prototype.undoTransaction = function(transaction) {
+    var sellerAccountInfo = UserService.currentUser().accountInfo,
+      customerAccountInfo = UserService.currentCustomer().accountInfo,
+      params = new RequestParameterBuilder()
+        .setOperationId('charge')
+        .setAgent(sellerAccountInfo.accountId)
+        .setMember(customerAccountInfo.accountId)
+        .setField('amount', transaction.amount)
+        .setField('description', transaction.description)
+        .setField('created', transaction.created)
+        .setField('force', -1)
+        .setField('goods', transaction.goods)
+        .getParams();
+
+    return this.makeRequest_(params, sellerAccountInfo)
+      .then(function(res) {
+        return res.data;
+      })
+      .then(function(transactionResult) {
```

Comment:
The user needs some way of knowing that this has been successful. Could you use the same success/error screen with the updated message? e.g. "The transaction of $0.25 was successfully reversed."